### ISSUE TITLE
[BUG] Falcon-40 have accuracy issue when using autotp

### ISSUE THREAD

**Issue by Yejing-Lai** [microsoft/DeepSpeed#4903]
**Describe the bug**
Falcon-40b autotp text-generation task will get the wrong output.

**To Reproduce**
1. Simple inference script to reproduce
[reproduce script](https://github.com/intel/intel-extension-for-pytorch/blob/main/examples/cpu/inference/python/llm/distributed/run_generation_with_deepspeed.py)
2. What packages are required and their versions
ipex: 2.1.100
torch: 2.1
oneccl 2.1.0
3. How to run the script
deepspeed --num_gpus 2 --bind_cores_to_rank run_generation_with_deepspeed.py --benchmark -m  tiiuae/falcon-40b

**Expected behavior**
input: DeepSpeed is a machine learning framework for deep neural networks (DNNs) and deep reinforcement learning (DRL). It is written in C++ and is
expected output: DeepSpeed is a machine learning framework for deep neural networks (DNNs) and deep reinforcement learning (DRL). It is written in C++ and is based on PyTorch.DeepSpeed is designed to scale DNNs and DRL to large numbers of GPUs and TPUs.
but get the wrong output: DeepSpeed is a machine learning framework for deep neural networks (DNNs) and deep reinforcement learning (DRL). It is written in C++ and is e e e e e e e e e e e e e e e e e e e e e e e e e e e e e e

**ds_report output**
DeepSpeed C++/CUDA extension op report
--------------------------------------------------
NOTE: Ops not installed will be just-in-time (JIT) compiled at
      runtime if needed. Op compatibility means that your system
      meet the required dependencies to JIT install the op.
--------------------------------------------------
JIT compiled ops requires ninja
ninja .................. [OKAY]
--------------------------------------------------
op name ................ installed .. compatible
--------------------------------------------------
deepspeed_not_implemented  [NO] ....... [OKAY]
deepspeed_ccl_comm ..... [NO] ....... [OKAY]
cpu_adam ............... [NO] ....... [OKAY]
fused_adam ............. [NO] ....... [OKAY]
--------------------------------------------------
DeepSpeed general environment info:
torch install path ............... ['/xxx/python3.9/site-packages/torch']
torch version .................... 2.1.2+cpu
deepspeed install path ........... ['/xxx/deepspeed']
deepspeed info ................... 0.12.7+83427253, 83427253, master
deepspeed wheel compiled w. ...... torch 2.1

**Comment by delock** [comment]
@Yejing-Lai I saw you have an PR https://github.com/microsoft/DeepSpeed/pull/4895, is this PR intended to fix this issue?
